### PR TITLE
 Generate coin metadata methods

### LINKED ISSUE #370: Coin configuration JSON
All coin configuration parameters should live in a JSON file. We can generate the C++ code from this file instead of this being a manual process for every coin.

### PR DESCRIPTION
**Pull request by alejandro-isaza**
All coin metadata methods are now generated from a JSON file.

Fix #370

### CODE REVIEW

File: coins.json [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+        "publicKeyHasher": "sha256ripemd",
+        "base58Hasher": "sha256d",
+        "xpub": "zpub",
+        "xprv": "zprv",
+        "transactionPath": "/tx/",
+        "explorer": "https://explorer.viacoin.org"
+    },
+    {
+        "id": "poa",
+        "name": "POA",
```
Comment (hewigovens) [inline coins.json]: POA Network

File: coins.json [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+        "decimals": 18,
+        "derivationPath": "m/44'/818'/0'/0/0",
+        "curve": "secp256k1",
+        "publicKeyType": "secp256k1Extended",
+        "transactionPath": "/transactions/",
+        "explorer": "https://explore.veforge.com"
+    },
+    {
+        "id": "wanchain",
+        "name": "Wanchain",
```
Comment (hewigovens) [inline coins.json]: let's keep WanChain

File: coins.json [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+        "decimals": 8,
+        "derivationPath": "m/44'/714'/0'/0/0",
+        "curve": "secp256k1",
+        "publicKeyType": "secp256k1",
+        "transactionPath": "/tx/",
+        "explorer": "https://explorer.binance.org"
+    },
+    {
+        "id": "ripple",
+        "name": "Ripple",
```
Comment (hewigovens) [inline coins.json]: It's XRP

File: coins.json (hunk 1 above)
Comment (alejandro-isaza) [reply coins.json]: Ok

File: coins.json (hunk 2 above)
Comment (alejandro-isaza) [reply coins.json]: They use Wanchain in their docs

File: coins.json (hunk 3 above)
Comment (alejandro-isaza) [reply coins.json]: that's the symbol, isn't it?

File: coins.json (hunk 2 above)
Comment (hewigovens) [reply coins.json]: oh ok

File: coins.json (hunk 3 above)
Comment (hewigovens) [reply coins.json]: XRP army complains, it's not Ripple

**Pull request merged**